app/app_iface.go

```diff
@@ -41,12 +41,22 @@ type AppIface interface {
 	ListAutocompleteCommands(teamID string, T i18n.TranslateFunc) ([]*model.Command, *model.AppError)
 	// @openTracingParams teamID, skipSlackParsing
 	CreateCommandPost(post *model.Post, teamID string, response *model.CommandResponse, skipSlackParsing bool) (*model.Post, *model.AppError)
+	// AddChannelMember adds a user to a channel. It is a wrapper over AddUserToChannel.
+	// checkTeamMember is used to indicate whether it will check for the user was deleted from
+	// the team or not. This is useful to avoid in scenarios when we just added the team member,
+	// and thereby know that there is no need to check this.
+	AddChannelMember(userID string, channel *model.Channel, userRequestorId string, postRootId string, checkTeamMember bool) (*model.ChannelMember, *model.AppError)
```

Comment:
5 feel like a lot of parameters. I'm thinking this (and `AddUserToChannel`) could take a struct with a boolean field where the default `false` value follows the higher data integrity code path. Maybe something like `unsafeTeamMemberDataIntegrity`. That way if you're using the method you can ignore that property in most cases, and if you don't ignore it you see "unsafe" and don't set it to `true` unless you look into it and know what you're doing.

Reply:
I like it. :+1:

Reply:
I refactored `AddChannelMember`. But kept `AddUserToChannel` as-is. It didn't seem like a huge benefit changing one parameter to a struct. But let me know if you want me to do that too. 0/5.